homeassistant/components/ozw/config_flow.py

```diff
@@ -1,24 +1,174 @@
+
+        return await self.async_step_install_addon()
+
+    async def async_step_install_addon(self):
+        """Install add-on."""
+        try:
+            await self.hass.components.hassio.async_install_addon("core_zwave")
+        except self.hass.components.hassio.HassioAPIError as err:
+            _LOGGER.error("Failed to install OpenZWave add-on: %s", err)
+            return self.async_abort(reason="addon_install_failed")
+        self.integration_created_addon = True
+
+        return await self.async_step_start_addon()
+
+    async def async_step_start_addon(self, user_input=None):
+        """Ask for config and start add-on."""
+        if self.addon_config is None:
+            self.addon_config = await self._async_get_addon_config()
+
+        errors = {}
+
         if user_input is not None:
-            return self.async_create_entry(title=TITLE, data={})
+            self.network_key = user_input[CONF_NETWORK_KEY]
+            self.usb_path = user_input[CONF_USB_PATH]
+
+            new_addon_config = {CONF_ADDON_DEVICE: self.usb_path}
+            if self.network_key:
+                new_addon_config[CONF_ADDON_NETWORK_KEY] = self.network_key
+
+            if new_addon_config != self.addon_config:
+                await self._async_set_addon_config(new_addon_config)
+
+            try:
+                await self.hass.components.hassio.async_start_addon("core_zwave")
+            except self.hass.components.hassio.HassioAPIError as err:
+                _LOGGER.error("Failed to start OpenZWave add-on: %s", err)
+                errors["base"] = "addon_start_failed"
+            else:
+                return self._async_use_mqtt_integration()
+
+        self.usb_path = self.addon_config.get(CONF_ADDON_DEVICE, "")
+        self.network_key = self.addon_config.get(CONF_ADDON_NETWORK_KEY, "")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USB_PATH, default=self.usb_path): str,
+                vol.Optional(CONF_NETWORK_KEY, default=self.network_key): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="start_addon", data_schema=data_schema, errors=errors
+        )
+
+    async def _async_get_addon_info(self):
+        """Return and cache add-on info."""
+        try:
+            addon_info = await self.hass.components.hassio.async_get_addon_info(
+                "core_zwave"
+            )
+        except self.hass.components.hassio.HassioAPIError as err:
+            _LOGGER.error("Failed to get OpenZWave add-on info: %s", err)
+            raise AbortFlow("addon_info_failed") from err
+
+        return addon_info
+
+    async def _async_is_addon_running(self):
+        """Return True if add-on is running."""
+        addon_info = await self._async_get_addon_info()
+        return addon_info["state"] == "started"
+
+    async def _async_is_addon_installed(self):
+        """Return True if add-on is installed."""
```

Comment:
comment mostly restates method name. which add-on?

Reply:
On line 78 we specify the OpenZWave add-on. That method has the entry point for all management of the add-on. I don't think we need to update all the add-on methods.

Reply:
👍 yep, sorry... that is clear!